### PR TITLE
Jomarton/checkbox accessibility

### PR DESCRIPTION
**Pull request by JoMarton**
This PR is addressing accessibility features in the checkbox component.
While reviewing requirements it was decided to change the initial approach and use the native input element as the "core" of the component to gain the native behaviors

### CODE REVIEW

File: src/components/checkbox/checkbox.st.css [hunk 1]
```diff
@@ -22,9 +22,13 @@
...
 }
 
+.root:focused .boxIconDefault {
+    outline: 4px solid #add9ff;
+}
+
 .tickIcon {
-    height: 1em;
-    width: 1em;
+    height: 16px;
```
Comment (AviVahl) [inline src/components/checkbox/checkbox.st.css]: em

File: src/components/checkbox/checkbox.tsx [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+}
+
+export interface CheckBoxProps extends React.HTMLAttributes<HTMLInputElement> {
     value: boolean;
     boxIcon: React.ComponentType<CheckBoxIconProps>;
     tickIcon: React.ComponentType<CheckBoxIconProps>;
     indeterminateIcon: React.ComponentType<CheckBoxIconProps>;
-    onChange: (value: boolean) => void;
+    onChange: (event: CheckBoxChangeEvent) => void;
     children?: any;
```
Comment (AviVahl) [inline src/components/checkbox/checkbox.tsx]: any?

File: test/components/checkbox.spec.tsx [hunk 3]
```diff
@@ -160,23 +160,62 @@ describe('<Checkbox/>', function() {
...
+    describe('Accessibility features', function() {
+        it('Renders a native input and pass on checked state', function() {
+            const {select, waitForDom} = clientRenderer.render(
+                <CheckBox value={true}/>
+            );
+            const nativeInput = select('NATIVE_CHECKBOX') as HTMLInputElement;
+
+            return waitForDom(() => {
+                expect(nativeInput, 'native input not found in DOM').to.exist;
+                expect(nativeInput.tagName).to.equal('INPUT');
```
Comment (AviVahl) [inline test/components/checkbox.spec.tsx]: instanceOf

File: test/components/checkbox.spec.tsx [hunk 4]
```diff
@@ -160,23 +160,62 @@ describe('<Checkbox/>', function() {
...
-        const nativeInput = select('NATIVE_CHECKBOX') as HTMLInputElement;
+    describe('Accessibility features', function() {
+        it('Renders a native input and pass on checked state', function() {
+            const {select, waitForDom} = clientRenderer.render(
+                <CheckBox value={true}/>
+            );
+            const nativeInput = select('NATIVE_CHECKBOX') as HTMLInputElement;
+
+            return waitForDom(() => {
+                expect(nativeInput, 'native input not found in DOM').to.exist;
```
Comment (AviVahl) [inline test/components/checkbox.spec.tsx]: present()

File: test/components/checkbox.spec.tsx [hunk 5]
```diff
@@ -160,23 +160,62 @@ describe('<Checkbox/>', function() {
...
+                expect(nativeInput.tagName).to.equal('INPUT');
+                expect(nativeInput).to.have.attr('type', 'checkbox');
+                expect(nativeInput.checked, 'native checkbox should be checked').to.equal(true);
```
Comment (AviVahl) [inline test/components/checkbox.spec.tsx]: same as line above?

File: test/components/checkbox.spec.tsx [hunk 6]
```diff
@@ -160,23 +160,62 @@ describe('<Checkbox/>', function() {
...
+            });
+        });
+
+        it('native input gets disabled state', function() {
+            const {select, waitForDom} = clientRenderer.render(
+                <CheckBox disabled/>
+            );
+            const nativeInput = select('NATIVE_CHECKBOX') as HTMLInputElement;
+            return waitForDom(() => {
+                expect(nativeInput.disabled, 'native checkbox should be disabled').to.equal(true);
```
Comment (AviVahl) [inline test/components/checkbox.spec.tsx]: same

File: test/components/checkbox.spec.tsx [hunk 7]
```diff
@@ -160,23 +160,62 @@ describe('<Checkbox/>', function() {
...
-        );
-        const nativeInput = select('NATIVE_CHECKBOX') as HTMLInputElement;
+    describe('Accessibility features', function() {
+        it('Renders a native input and pass on checked state', function() {
+            const {select, waitForDom} = clientRenderer.render(
+                <CheckBox value={true}/>
+            );
+            const nativeInput = select('NATIVE_CHECKBOX') as HTMLInputElement;
+
+            return waitForDom(() => {
```
Comment (AviVahl) [inline test/components/checkbox.spec.tsx]: async/await

File: src/components/checkbox/checkbox.tsx (hunk 2 above)
Comment (JoMarton) [inline src/components/checkbox/checkbox.tsx]: ReactNode

File: test/components/checkbox.spec.tsx (hunk 5 above)
Comment (JoMarton) [inline test/components/checkbox.spec.tsx]: the `.checked` attribute doesn't appear in the dom so the `.attribute` matcher doesn't work the same way

**Pull request merged**